### ISSUE TITLE
[Repo] Edit PR template to include review checklist

### ISSUE THREAD

**Issue by davidicus** [carbon-design-system/carbon-addons-iot-react#2426]
We need to add a checklist to remind reviewers of the things they should be looking for when approving a PR.

- RTL check of UI using storybook knob
- Check the a11y test has been written and no violations exist. (can verify with the a11y knob)
- Are all strings translatable 
- are `data-testid`s in place in containing component and major areas of the UI
- are there visual regression test written for the component. (once we have visual testing framework in place)
